File: Bubble_Sort.cpp
#include <iostream>

void swap(int* a, int* b) {
    int t = *a;
    *a = *b;
    *b = t;
}

void bubbleSort(int arr[], int n) {
    bool swapped;
    for (int i = 0; i < n - 1; i++) {
        swapped = false;
        for (int j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(&arr[j], &arr[j + 1]);
                swapped = true;
            }
        }
        if (swapped == false) {
            break;
        }
    }
}

void printArray(int arr[], int size) {
    for (int i = 0; i < size; i++)
        std::cout << arr[i] << " ";
    std::cout << std::endl;
}

int main() {
    int arr[] = {9, 14, 95, 12, 42, 51, 90};
    int n = sizeof(arr) / sizeof(arr[0]);
    std::cout << "Unsorted Array:\n";
    printArray(arr,n);
    bubbleSort(arr, n);
    std::cout << "Sorted array in ascending order: \n";
    printArray(arr, n);
}

File: Bubble_Sort.js
function swap(arr, i, j) {
    let t = arr[i];
    arr[i] = arr[j];
    arr[j] = t;
}

function bubbleSort(arr) {
    let n = arr.length;
    let swapped = true;
    for (let i = 0; i < n - 1; i++) {
        swapped = false;
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(arr, j, j + 1);
                swapped = true;
            }
        }
        if (!swapped) {
            break;
        }
    }
}

function printArray(arr) {
    for (let i = 0; i < arr.length; i++) {
        console.log(arr[i], end=" ");
    }
    console.log();
}

let arr = [9, 14, 95, 12, 42, 51, 90];
console.log("Unsorted Array:");
printArray(arr);
bubbleSort(arr);
console.log("Sorted array in ascending order:");
printArray(arr);